src/services/UiApi.ts

```diff
@@ -34,4 +44,16 @@ export class UiApi extends WorkspaceApi implements IUiApi {
 
     return data
   }
+
+  public async getNextRunsByFlow(flowIds: string[]): Promise<UiNextFlowRunByFlow> {
+    const request = { 'flow_ids': [...new Set(flowIds)] }
```

Comment:
Do we need to dedupe this? Creating sets is slow. Probably not a big deal since this shouldn't get called many times. But wouldn't the api dedupe this as well in its query?

Reply:
I think the API definitely will handle that - the case I was solving for here was when multiple components for the same flow id made the request - my thinking was that deduping to start could save us round-trips when we run into limits on the batcher. E.g. If i make 201 calls (batcher limits 200) and the 199th and 201st call are the same ID. Obviously a contrived scenario but I could see us running into it pretty easily

Reply:
Oh but actually now that I walk the code.. this deduping would need to happen at the batch level and not at the route level 🤦🏻